File: hackerearth/golang/the-minimum-cost.go
/**
 * https://www.hackerearth.com/practice/basic-programming/bit-manipulation/basics-of-bit-manipulation/practice-problems/algorithm/min-cost-2-fe2d3308/
 **/
package main

import "fmt"

func solveTestCaseTheMimimumCost() {
	var n, c01, c10, costA, costB int64
	fmt.Scanf("%d %d %d", &n, &c01, &c10)

	bits := make([]int, n)
	for i := int64(0); i < n; i++ {
		fmt.Scanf("%d", &bits[i])
	}

	a, b := 0, 1
	costA = 0
	costB = 0
	for _, bit := range bits {
		if bit != a {
			costA += int64(bit^1)*c01 + int64(bit^0)*c10
		}

		if bit != b {
			costB += int64(bit^1)*c01 + int64(bit^0)*c10
		}
		a = a ^ 1
		b = b ^ 1
	}

	ans := costB
	if costA < costB {
		ans = costA
	}
	fmt.Println(ans)
}

func theMinimumCost() {
	var t int
	fmt.Scanf("%d", &t)

	for t > 0 {
		solveTestCaseTheMimimumCost()
		t--
	}
}

File: hackerearth/java/TheMinimumCost.java
/**
 * https://www.hackerearth.com/practice/basic-programming/bit-manipulation/basics-of-bit-manipulation/practice-problems/algorithm/min-cost-2-fe2d3308/
 **/
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

class TheMinimumCost {
    static Scanner scanner = new Scanner(System.in);
    
    static void solve_test_case() {
        scanner.nextLine();
        long n = scanner.nextInt();
        long c01 = scanner.nextInt();
        long c10 = scanner.nextInt();
        int b;
        scanner.nextLine();
        List<Integer> bits = new ArrayList<>((int)n);
        for (int i = 0; i < n; i++) {
            b = scanner.nextInt();
            bits.add(b);
        }

        long cost_a=0;
        int a = 0;
        b = 1;
        long cost_b=0;
        for(Integer bit: bits) {
            if (bit != a) {
                cost_a += bit == 1 ? c10: c01;
            }
            if (bit != b) {
                cost_b += bit == 1 ? c10: c01;
            }
            a ^= 1;
            b ^= 1;
        }
        System.out.println(cost_a < cost_b? cost_a: cost_b);
    }
    public static void main(String[] args) throws Exception {
        int t = scanner.nextInt();
        while (t > 0) {
            solve_test_case();
            t--;
        }
        scanner.close();
    }
}
